gallery/entrypoint.sh

```diff
@@ -0,0 +1,166 @@
+    fi
+
+    echo "Package: ${PROJECT}"
+    echo "Black versions: ${BLACK_VERSIONS[*]}"
+
+    echo "------------------------------------"
+    read -p "Continue? " -n 1 -r
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+        echo " Aborting."
+        exit 1
+    fi
+    echo # newline, for aesthetics
+}
+
+get_source_for_project() {
+    project="$1"
+    version="$2"
+    target="${TARGET}${project}"
+
+    [ -e "${target}" ] && echo "[WARNING] Path already exists: ${target}"
+    mkdir -p "${target}"
+
+    case "${project}" in
+    cpython)
+        version="${version:-master}" # assume "master" if version not specified
+        src="https://github.com/python/cpython/archive/${version}.zip"
+        echo "getting source: ${src}"
+        wget "${src}" -P /tmp
+        unzip -qd "${target}" "/tmp/${version}.zip"
+        ;;
+    mypy)
+        version="${version:-branch/default}" # assume "branch/default" if version not specified
+        src="https://foss.heptapod.net/pypy/pypy/repository/${version}/archive.tar.bz2"
+        echo "getting source: ${src}"
+        wget -O- "${src}" | tar -xjf - -C "${target}"
+        ;;
+    *)
+        if [ ! -z "${version}" ]; then
+            project="${project}==${version}"
+        fi
+        echo "Running: pip install --quiet --no-warn-script-location --no-deps --no-binary all --no-cache '--prefix=${target}' '${project}'"
+        pip install --quiet --no-warn-script-location --no-deps --no-binary all --no-cache "--prefix=${target}" "${project}"
+        ;;
+    esac
+
+    echo "${target}"
+}
+
+get_black() {
+    version=${1:-master} # Assume "master" if version not provided
+
+    if [[ $version == "master" ]]; then
+        src="git+https://github.com/psf/black.git"
+    else
+        src="Black==${version}"
+    fi
+
+    echo "Installing ${src}"
+    pip install --quiet --force-reinstall "${src}"
+}
+
+get_config_str() {
+    toml="$1"
+    if [ ! -z "${toml}" ]; then
+        if [ ! -f "${toml}" ]; then
+            echo >&2 "[ERROR] You specified file ${toml} but it doesn't exist, did you mount it?"
+            echo >&2 "directory listing:"
+            ls >&2 -la
+            exit 1
+        else
+            echo "--config ${toml}"
+        fi
+    fi
+}
+
+init_git() {
+    cd "${1}"
+    echo "Initializing git repository in ${1}."
+    if [ -d ".git" ]; then
+        echo "[WARNING] .git directory already exists."
+    fi
+
+    git init .
+    git add .
+    git commit -q -m "Initial commit"
+}
+
+main() {
+    for bv in "${BLACK_VERSIONS[@]}"; do
+
+        # extract .toml file if present
+        IFS=':'
+        read -ra PAIR <<<"${bv}" # split version string using ':' as delimiter
+        if [ ${#PAIR[@]} -gt 1 ]; then
+            bv=${PAIR[0]}
+            toml=${PAIR[1]}
+            echo "black version: ${bv}"
+            echo "requested toml: ${toml}"
+        fi
+        IFS=' '
+
+        config=$(get_config_str "${toml}")
+        init_git "${TARGET}"
```

Comment:
We want one Git repository for all branches. This way we will be able to easily compare branches with `git diff` to see how different versions of Black format the same code.